src/execution/operator/projection/physical_range.cpp

```diff
@@ -0,0 +1,175 @@
+#include "duckdb/execution/operator/projection/physical_range.hpp"
+
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/common/algorithm.hpp"
+#include "duckdb/execution/expression_executor.hpp"
+#include "duckdb/planner/expression/bound_reference_expression.hpp"
+
+namespace duckdb {
+
+//===--------------------------------------------------------------------===//
+// Range (timestamp)
+//===--------------------------------------------------------------------===//
+
+class RangeOperatorState : public OperatorState {
```

Comment:
I'm fine keeping all this logic, but don't inherit from OperatorState and change the name of this class to avoid confusion.